lib/ddtrace/sampling/rule_sampler.rb

```diff
@@ -53,7 +53,7 @@ def initialize(rules = [],
                              # because it breaks its current contract to existing users.
                              Datadog::RateSampler.new.tap { |s| s.sample_rate = default_sample_rate }
                            else
-                             Datadog::AllSampler.new
+                             RateByServiceSampler.new(1.0, env: -> { Datadog.tracer.tags[:env] })
```

Comment:
will this get updates rates from the agent when we flush?

Reply:
Yes, we forward calls to `#update` to the `@default_sampler`: https://github.com/DataDog/dd-trace-rb/blob/dbcac580b0bbf314901ded18be7143eaaa5265cb/lib/ddtrace/sampling/rule_sampler.rb#L78

Reply:
🙌